client/hotrod-client/src/main/java/org/infinispan/client/hotrod/counter/impl/NotificationManager.java

```diff
@@ -178,12 +177,10 @@ public void stop() {
       log.debugf("Stopping %s (%s)", this, lock);
       lock.lock();
       try {
-         clientListeners.clear();
-         // Rather than de-registering all the listeners we'll terminate the channel and let server deal with it
-         if (dispatcher != null) {
-            dispatcher.close();
-            dispatcher = null;
-         }
+         CompletableFuture[] futures = clientListeners.keySet().stream().map(counterName ->
```

Comment:
Is it an issue that `clientListeners` isn't cleared any more?

Reply:
IDK, I have added the clear and the testsuite didn't break...